main.py

```diff
@@ -59,7 +59,9 @@ async def create_subscription(event_type, event_params):
             except (websockets.exceptions.ConnectionClosed, websockets.exceptions.ConnectionClosedError) as e:
                 logger.debug(f"Connection closed: {e}. Reconnecting...")
                 await asyncio.sleep(5)  # Wait for 5 seconds before reconnecting
-            except Exception as e:  # Catch all other exceptions, including asyncio.CancelledError
+            # asyncio.CancelledError is not a subclass of Exception starting Py3.8, see https://docs.python.org/3.11/library/asyncio-exceptions.html#asyncio.CancelledError
+            # Note that there might be more elecant solutions than to simply catch all BaseExceptions, but I would see this rather as a question on the websocket handling, nothing PTB related
```

Comment:
Thanks for the comments! I understand, so even if it's not related to PTB, I still don't quite understand why `asyncio.CancelledError` is thrown after an exception is catch in websocket error.

Shouldn't `except Exception as e` catch any error related to the websocket?

Reply:
Well, except Exception will catch only errors that derive from Exception, not from Base exception, that's the issue here

Why CancelledError is raised from the internals of websockets is rather out of scope for me :D